### ISSUE TITLE
Connection Call too verbose

### ISSUE THREAD

**Issue by MrCaracara** [SAP/PyRFC#176]
I have an application that uses PyRFC to retrieve some data from SAP. This application is deployed as a (Linux) Docker container, and I have tested it locally on a Windows computer.

When running on Windows, as well as on the Linux container up to very recently, everything worked fine. However, out of nowhere (there were no changes to the code), when running on the container, every call to SAP using PyRFC prints all of the returned data on the screen, cluttering the Docker logs with unnecessary (and **confidential!**) information.

Does anyone have any idea as to what could have changed and how to fix this?

**Comment by bsrdjan** [comment]
Could you please post the pyrfc version, docker base image and sap nwrfc SDK version?

**Comment by MrCaracara** [comment]
* __PyRFC Version__: 2.0.5 (Dockerfile pulls the current master branch of this repo)
* __Base Image__: python:3.7-buster
* __Docker__: Docker Desktop 2.3.0.2 (45183)
* __nwrfcsdk__: Unsure how to check this, but they were downloaded on April 27, so whichever was the latest version on that date.

**Comment by bsrdjan** [comment]
I created following docker container

**Dockerfile**

```docker
# docker build -t buster .
# docker run --name buster buster

FROM python:3.7-buster

# sap nw rfc lib
ARG nwrfc_local=/usr/local/sap
ENV SAPNWRFC_HOME ${nwrfc_local}/nwrfcsdk
RUN printf "\n# nwrfc sdk \n" >> ~/.bashrc
RUN printf "export SAPNWRFC_HOME=${nwrfc_local}/nwrfcsdk \n" >> ~/.bashrc
RUN mkdir -p ${nwrfc_local}
COPY /SAP/NWRFCSDK-750-PL6/linux/nwrfcsdk ${nwrfc_local}/nwrfcsdk
RUN chmod -R a+rx ${nwrfc_local}/nwrfcsdk/lib*
RUN printf "# include nwrfcsdk\n${nwrfc_local}/nwrfcsdk/lib\n" | tee /etc/ld.so.conf.d/nwrfcsdk.conf
RUN ldconfig && ldconfig -p | grep sap

COPY testscript.py ./testscript.py

RUN pip install pynwrfc

CMD ["python", "./testscript.py"]
```

and used this script for testing:

**testscript.py**

```python
from pyrfc import Connection

abap_system = {
    "user": "demo",
    "passwd": "welcome",
    "ashost": "10.68.110.51",
    "sysnr": "00",
    "client": "620",
    "lang": "EN",
}

client = Connection(**abap_system)

# should be visible in log
print("version", client.version)
print("connection", client.get_connection_attributes())

# should not be visible in log
result = client.call("STFC_CONNECTION", REQUTEXT="hello")

# should be visible in log
print("bye!")
```

Running the docker script shows:

```shell
$ docker run --name buster buster
version {'major': 7500, 'minor': 0, 'patchLevel': 6}
connection {'dest': '', 'host': '8c6df1d7a62f', 'partnerHost': 'coevi51', 'sysNumber': '00', 'sysId': 'MME', 'client': '620', 'user': 'DEMO', 'language': 'E', 'trace': '0', 'isoLanguage': 'EN', 'codepage': '4103', 'partnerCodepage': '4103', 'rfcRole': 'C', 'type': 'E', 'partnerType': '3', 'rel': '753', 'partnerRel': '750', 'kernelRel': '745', 'cpicConvId': '82673798', 'progName': 'SAPLSYST', 'partnerBytesPerChar': '2', 'partnerSystemCodepage': '4103', 'partnerIP': '10.68.110.51', 'partnerIPv6': '10.68.110.51', 'active_unit': False}
bye!

$ docker logs --details buster
 version {'major': 7500, 'minor': 0, 'patchLevel': 6}
 connection {'dest': '', 'host': '8c6df1d7a62f', 'partnerHost': 'coevi51', 'sysNumber': '00', 'sysId': 'MME', 'client': '620', 'user': 'DEMO', 'language': 'E', 'trace': '0', 'isoLanguage': 'EN', 'codepage': '4103', 'partnerCodepage': '4103', 'rfcRole': 'C', 'type': 'E', 'partnerType': '3', 'rel': '753', 'partnerRel': '750', 'kernelRel': '745', 'cpicConvId': '82673798', 'progName': 'SAPLSYST', 'partnerBytesPerChar': '2', 'partnerSystemCodepage': '4103', 'partnerIP': '10.68.110.51', 'partnerIPv6': '10.68.110.51', 'active_unit': False}
 bye!
```

Could you please share the Dockerfile, test script and the logs, like above, reproducing the issue? You could use standard ABAP RFC calls, like in my test, or `STFC_STRUCTURE`, like in [unit test](https://github.com/SAP/PyRFC/blob/master/tests/test_connection.py#L178).

**Comment by MrCaracara** [comment]
I copied your __testscript.py__ (only modified the connection parameters) and modified my Dockerfile to execute it:

__Dockerfile__
```
FROM python:3.7-buster

RUN apt-get update

# Add files to image
ADD src /src
ADD nwrfcsdk /usr/local/sap/nwrfcsdk

# Install NWRFCSDK
ENV SAPNWRFC_HOME=/usr/local/sap/nwrfcsdk
RUN echo "/usr/local/sap/nwrfcsdk/lib" >> /etc/ld.so.conf.d/nwrfcsdk.conf
RUN ldconfig

# Install PyRFC
WORKDIR /tmp
RUN git clone https://github.com/SAP/PyRFC.git
RUN cd PyRFC \
 && python setup.py clean --all \
 && python setup.py bdist_wheel \
 && pip install dist/pynwrfc-2.0.5-cp37-cp37m-linux_x86_64.whl

# Configure Python Program
WORKDIR /src
ENTRYPOINT ["python", "testscript.py"]
```

The SAP account I'm using lacks access to `STFC_CONNECTION`, but you can see the issue in the output anyway:

```
version {'major': 7500, 'minor': 0, 'patchLevel': 6}
49 48 48 0 83 chars 3 bytes 10 rlen 3 utf8 b'100'
83 65 45 69 84 chars 12 bytes 37 rlen 12 utf8 b'SA-ETL-LIM'
69 0 0 48 0 chars 2 bytes 7 rlen 2 utf8 b'E'
connection {'client': '100', 'user': 'SA-ETL-LIM', 'language': 'E', 'active_unit': False}
78 111 32 82 70 chars 57 bytes 172 rlen 57 utf8 b'No RFC authorization for function module STFC_CONNECTION.'
82 70 67 95 78 chars 16 bytes 49 rlen 16 utf8 b'RFC_NO_AUTHORITY'
Traceback (most recent call last):
  File "testscript.py", line 18, in <module>
    result = client.call("STFC_CONNECTION", REQUTEXT="hello")
  File "src/pyrfc/_pyrfc.pyx", line 433, in pyrfc.pyrfc.Connection.call
  File "src/pyrfc/_pyrfc.pyx", line 259, in pyrfc.pyrfc.Connection._error
pyrfc._exception.ABAPRuntimeError: RFC_ABAP_RUNTIME_FAILURE (rc=3): key=RFC_NO_AUTHORITY, message=No RFC authorization for function module STFC_CONNECTION. [MSG: class=, type=, number=, v1-4:=;;;]
```

I assume that the expected output would be:

```
version {'major': 7500, 'minor': 0, 'patchLevel': 6}
connection {'client': '100', 'user': 'SA-ETL-LIM', 'language': 'E', 'active_unit': False}
Traceback (most recent call last):
  File "testscript.py", line 18, in <module>
    result = client.call("STFC_CONNECTION", REQUTEXT="hello")
  File "src/pyrfc/_pyrfc.pyx", line 433, in pyrfc.pyrfc.Connection.call
  File "src/pyrfc/_pyrfc.pyx", line 259, in pyrfc.pyrfc.Connection._error
pyrfc._exception.ABAPRuntimeError: RFC_ABAP_RUNTIME_FAILURE (rc=3): key=RFC_NO_AUTHORITY, message=No RFC authorization for function module STFC_CONNECTION. [MSG: class=, type=, number=, v1-4:=;;;]

```

As you can see, all the data that PyRFC returns is being directly sent to stdout before we even print it.

**Comment by MrCaracara** [comment]
I also tried adding a `"trace"` parameter to the connection as you mentioned.
I tried with 0, 1, 2 and 3, and there was no changes in the output the lines in the style of:
`49 48 48 0 83 chars 3 bytes 10 rlen 3 utf8 b'100'`
were still being printed out.

**Comment by bsrdjan** [comment]
It was the bug in 2.0.5, first 5 characters of ABAP strings were echoed.

The fix will be shipped with next release and in the meantime please use the  build from source from current master.

**Comment by MrCaracara** [comment]
Ah, yes that was the problem. I see there was a print statement that was not commented. Your last commit fixed this issue and now it works correctly.

Thanks a lot, I will close the issue now, as it has been resolved.

**Comment by amarvin** [comment]
The commit 6cdccbafacb38e6ceec3e3ed4770a9e66b1d1abd

**Comment by bsrdjan** [comment]
fixed in [2.0.6](https://github.com/SAP/PyRFC/releases/tag/2.0.6)
